01.md

```diff
@@ -35,6 +35,10 @@ Once `LNURL` is decoded:
 
 Neither status codes or any HTTP Header has any meaning. Servers may use whatever they want. Clients should ignore them (and be careful when using libraries that treat responses differently based on headers and status codes) and just parse the response body as JSON, then interpret it accordingly.
 
+## CORS Browser Compatibility
+
+Application endpoints (or their defined Reverse Proxy directives) accessed when using the LNURL must respond to `GET` having set the header `Access-Control-Allow-Origin: *` for LNURL's to work with browser-based wallets, services and other webapps. If they set the `Access-Control-Allow-Methods` header, they must set in such a way that `GET` requests are permitted.
```

Comment:
```suggestion
Application endpoints or Reverse Proxy directives that are accessed when using the LNURL must respond to `OPTIONS` requests on that endpoint. In their replies to `GET` and `OPTIONS`, they must set the header `Access-Control-Allow-Origin: *` for LNURL's to work with browser-based wallets, services and other webapps. If they set the `Access-Control-Allow-Methods` header, they must set in such a way that `GET` requests are permitted.
```
```suggestion
Application endpoints that need to be accessed when interacting with an LNURL must set CORS headers in such a way that they are reachable from any web origin.
```

Based on feedback from @hsjoberg and @fiatjaf, maybe just this sentence would be best.

No someone else apart from me and @shocknet-justin should probably also decide if we should include reverse proxies, because we don't seem to be able to agree here.

Reply:
I think we should at least keep the `Access-Control-Allow-Origin: *` or/and a link to the [MDN describing CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS), to make it easier & more straightforward for devs to actually implement.

The issues is that incorrectly configured CORS is not immediately noticeable by a dev who uses non webapp based wallet (majority of wallets currently), I am worried that if this is not straightforward to implement (e.g. devs having to research what this CORS even means) they might just skip through it thinking their LN service is working all good.

Maybe it'd make sense to have a simple online tool (webapp) for checking whether LNURLs adhere to the spec (including CORS)

Reply:
@adambor I added your suggestion to my comment above.

@kaloudis @Kukks @adambor What's your opinion about mentioning proxies?

Reply:
I think we should remove the mention of proxies to keep the text shorter.

Isn't it already obvious for proxies that CORS header would be required? The proxy is really an "application endpoint" here if I understand it correctly.

Reply:
Feel like we're bikeshedding on the proxy text. On one hand it's not taking up much space, on the other it also feels obvious to me.

I could go either way, but I lean towards keeping it as is.

Reply:
Great! Two more people disagreeing 🙃

Reply:
I think it's obvious and an unnecessary implementation detail to mention.